File: bfs_dfs/bfs.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> v[10]; // adj matrix
int level[10]; // level of the graph
bool visited[10]; // node visited or not

void bfs(int s){
    queue<int> q;
    q.push(s);
    level[s] = 0; // setting the current root as the level 0
    visited[s] = true;

    while (!q.empty()){
        int p = q.front();
        q.pop();
        for (int i = 0; i < v[p].size(); i++){
            if (visited[v[p][i]] == false){
                level[v[p][i]] = level[p]+1;
                q.push(v[p][i]);
                visited[v[p][i]]=true;
            }
        }
    }
}

int main(){
    

    return 0;
}

File: bfs_dfs/bfs.py
from collections import defaultdict
from queue import Queue


def bfs(start, matrix):
    visited = [False]*(max(matrix)+1)
    que = Queue(n)

    que.put(start)
    visited[start] = True

    while not que.empty():
        start = que.get()
        print(start, end=" ")

        for node in matrix[start]:
            if visited[node] == False:
                que.put(node)
                visited[node] = True

if __name__ == "__main__":
    n = int(input())
    matrix = defaultdict(list)

    for _ in range(1,n+1):
        u, v = map(int, input().split())
        matrix[u].append(v)
        matrix[v].append(u)

    bfs(1, matrix)


